src/include/miopen/smoothl1loss/problem_description.hpp

```diff
@@ -0,0 +1,262 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/activ.hpp>
+#include <miopen/problem_description_base.hpp>
+#include <miopen/tensor.hpp>
+
+#include <cassert>
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace smoothl1loss {
+
+bool checkSameType(const TensorDescriptor& x, const TensorDescriptor& y);
+bool checkSameLength(const TensorDescriptor& x, const TensorDescriptor& y);
+bool checkSameStride(const TensorDescriptor& x, const TensorDescriptor& y);
+bool checkRightStride(const TensorDescriptor& x);
+bool checkContiguous(const TensorDescriptor& x);
+
+struct SmoothL1LossFwdProblemDescription : ProblemDescriptionBase
+{
+    SmoothL1LossFwdProblemDescription(const TensorDescriptor& iDesc_,
+                                      const TensorDescriptor& tDesc_,
+                                      const TensorDescriptor& oDesc_)
+        : iDesc(iDesc_), tDesc(tDesc_), oDesc(oDesc_)
+    {
+    }
+
+    const TensorDescriptor& GetIDesc() const { return iDesc; }
+    const TensorDescriptor& GetTDesc() const { return tDesc; }
+    const TensorDescriptor& GetODesc() const { return oDesc; }
+
+    bool IsSameType() const
+    {
+        if(!checkSameType(iDesc, tDesc))
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Reduce: Tensor types do not match.");
+#else
+            return false;
+#endif
+        }
+        return true;
+    }
+
+    bool IsRightLength() const
+    {
+        if(!checkSameLength(iDesc, tDesc))
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Smooth L1Loss: Tensor sizes do not match.");
+#else
+            return false;
+#endif
+        }
+        return true;
+    }
+
+    bool IsRightStride() const
+    {
+        if(!checkRightStride(iDesc) || !checkRightStride(tDesc) || !checkRightStride(oDesc))
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Smooth L1Loss: Tensor strides do not match.");
```

Comment:
The error msg here is quite confuse with err msg in IsSameStride() function

Reply:
Agree on this one. Please update those msgs, or one of them.